File: Model-View-ViewModel-Java/app/src/main/java/com/mvvm/java/core/di/module/ActivityModule.java
package com.mvvm.java.core.di.module;

import android.content.Context;

import androidx.appcompat.app.AppCompatActivity;

import com.mvvm.java.core.di.ActivityContext;
import com.mvvm.java.core.utilities.rx.SchedulerProvider;
import com.mvvm.java.core.utilities.rx.SchedulerProviderImplement;

import dagger.Module;
import dagger.Provides;
import io.reactivex.rxjava3.disposables.CompositeDisposable;

@Module
public class ActivityModule {

    private final AppCompatActivity activity;

    public ActivityModule(AppCompatActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return activity;
    }

    @Provides
    AppCompatActivity provideActivity() {
        return activity;
    }

    @Provides
    CompositeDisposable provideCompositeDisposable() {
        return new CompositeDisposable();
    }

    @Provides
    SchedulerProvider provideSchedulerProvider() {
        return new SchedulerProviderImplement();
    }
}

File: Android-MVP-Kotlin/app/src/main/java/com/base/mvp/core/di/modules/ActivityModule.kt
package com.base.mvp.core.di.modules

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import com.base.mvp.core.di.ActivityContext
import dagger.Module
import dagger.Provides
import io.reactivex.rxjava3.disposables.CompositeDisposable

/**
 * Author: William Giang Nguyen | 7/7/2022
 * */
@Module
class ActivityModule {

    private var activity: AppCompatActivity? = null

    fun ActivityModule(activity: AppCompatActivity?) {
        this.activity = activity
    }

    @Provides
    @ActivityContext
    fun provideContext(): Context? {
        return activity
    }

    @Provides
    fun provideActivity(): AppCompatActivity? {
        return activity
    }

    @Provides
    fun provideCompositeDisposable(): CompositeDisposable? {
        return CompositeDisposable()
    }
}